quetz/cli.py

```diff
@@ -0,0 +1,283 @@
+
+    return cleaned_deployements
+
+
+def _clean_deployements():
+
+    _ = _cleaned_deployements()
+
+
+@app.command()
+def create(path: str, 
+           config_file_name: str = "config.toml", 
+           create_conf: bool = False, 
+           test: bool = False):
+
+    abs_path = os.path.abspath(path)
+    config_file = os.path.join(path, config_file_name)
+    deployements = _get_deployements()
+
+    if os.path.exists(path):
+        if abs_path in deployements:
+            delete_ = typer.confirm('Quetz deployement exists at {}.\n'.format(path) +
+                                    'Overwrite it?')
+            if delete_:
+                delete(path, force=True)
+                create(path, config_file_name, create_conf, test)
+                return
+            else:
+                typer.echo('Use the start command to start a deployement.')
+                raise typer.Abort()
+
+        # only authorize path with a config file to avoid deletion of unexpected files
+        # when deleting Quetz instance
+        if not all([f in [config_file_name] for f in os.listdir(path)]):
+            typer.echo('Quetz deployement not allowed at {}.\n'.format(path) +
+                       'The path should not contain more than the configuration file.')
+            raise typer.Abort()
+
+        if not os.path.exists(config_file) and not create_conf:
+            typer.echo('Config file "{}" does not exist at {}.\n'.format(config_file_name,
+                                                                         path) + 
+                       'Use --create-conf option to generate a default config file.')
+            raise typer.Abort()
+    else:
+        if not create_conf:
+            typer.echo('No configuration file provided.\n' + 
+                       'Use --create-conf option to generate a default config file.')
+            raise typer.Abort()
+
+        Path(path).mkdir(parents=True)
+
+    if not os.path.exists(config_file):
+        conf = create_config()
+        with open(config_file, 'w') as f:
+            f.write(conf)
+
+    os.environ['QUETZ_CONFIG_FILE'] = config_file
+    load_configs(config_file)
+
+    os.chdir(path)
+    Path('channels').mkdir()
+    init_db()
+ 
+    if test:
+        _fill_test_database()
+
+    _store_deployement(abs_path, config_file_name)
+
+
+@app.command()
+def start(path: str, port: int = 8000, reload: bool = False):
+
+    abs_path = os.path.abspath(path)
+    deployements = _get_deployements()
+
+    try:
+        config_file_name = deployements[abs_path]
+    except KeyError:
+        typer.echo('No Quetz deployement found at {}.'.format(path))
+        raise typer.Abort()
+
+    config_file = os.path.join(abs_path, config_file_name)
+    os.environ[_env_prefix + _env_config_file] = config_file
+    os.chdir(path)
+
+    import quetz
+    quetz_src = os.path.dirname(quetz.__file__)
+    uvicorn.run("quetz.main:app", reload=reload, reload_dirs=(quetz_src, ), port=port)
```

Comment:
Would this tie people in to using uvicorn for their deployment? uvicorn's documentation for example recommends running via `gunicorn -k uvicorn.workers.UvicornWorker` for production, though other process managers are available.
This also misses parameters here which I and others may make use of, e.g. `--host` and `--proxy-headers`.

Reply:
It's a first shot, to be improved and complemented!
We can list missing options ? I will implement them!

> Would this tie people in to using uvicorn for their deployment?

This is not my intention, we can implement other managers and make it configurable?

Reply:
> We can list missing options ? I will implement them!

I think for now as long as `--host`, `--proxy-headers` and `--log-level` are implemented, I'm happy.